### PR TITLE
Add the file list page.

### PR DESCRIPTION
**Pull request by olha-khomiakova**
Render the file list page. Create components like this: FileListPage, UserDetails, Breadcrumbs, FileList, FileListItem, FolderListItem, Button, ButtonPanel, LinkWithAction.
All tests passed.
ESLint passed.

### CODE REVIEW

**Comment by dkalinin** [comment]
@olha-khomiakova please fix the conflicts.

File: js/app/file-list-page/file-list-item/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+   * @private
+   */
+  _convertSize() {
+    if (this.size > 1073741824) {
+      return Math.round(this.size/1073741824) + 'GB';
+    } else if (this.size > 1048576) {
+      return Math.round(this.size/1048576) + 'MB';
+    } else {
+      return Math.round(this.size/1024) + 'KB';
+    }
```
Comment (dkalinin) [inline js/app/file-list-page/file-list-item/index.js:58]: Let's show `0 B` or `5 B` if size is less than 1024.

File: js/app/file-list-page/file-list/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   * @param {ItemJson} firstItem
+   * @param {ItemJson} secondItem
+   * @returns {number} Returns -1 if the type of the first element is "file"
+   * and the type of the second element is "folder"
+   * or if their types are the same then they will be compared by their names.
+   * Otherwise 1 is returned.
+   * @private
+   */
+  _sortByTypeAndName(firstItem, secondItem) {
+    if (firstItem.constructor.name === secondItem.constructor.name) {
```
Comment (dkalinin) [inline js/app/file-list-page/file-list/index.js:36]: Let's use type field instead.

File: js/app/file-list-page/link-with-action/index.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Component that renders a LinkWithActions.
+ */
+export class LinkWithActions extends Component {
+  /**
+   * Adds an event for the link.
+   * @param {Function} handler
+   */
+  addEvent(handler) {
+    this.handller= handler;
```
Comment (dkalinin) [inline js/app/file-list-page/link-with-action/index.js:12]: Please reformat code.

File: js/app/file-list-page/link-with-action/index.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  /**
+   * Sets classes for icon.
+   * @param {String} classes are a string with set of name of classes.
+   */
+  setClassesForIcon(classes) {
+    const glyphicon = document.createElement('span');
+    classes.split(' ').forEach((oneClass)=>{
+      glyphicon.classList.add(oneClass);
+    });
```
Comment (dkalinin) [inline js/app/file-list-page/link-with-action/index.js:48]: Let's use https://developer.mozilla.org/en-US/docs/Web/API/Element/className which doesn't need the class string to be split by spaces.

**Review by dkalinin — CHANGES_REQUESTED** [review]
PTAL

File: js/app/components/button/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  createIcon(...classes) {
+    const icon = this.getElement('icon');
+    classes.forEach((oneClass)=>icon.classList.add(oneClass));
+  }
+
+  /**
+   * Adds an event on the button.
+   * @param {Function} handler
+   */
+  addEvent(handler) {
```
[the remark this answers is not in the document]
Comment (olha-khomiakova) [reply js/app/components/button/index.js:28]: Done.

File: js/app/components/button/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    this.handller= handler;
+  }
+  /**
+   * @inheritDoc
+   * @param {Function} handler
+   */
+  addEventListeners(handler) {
+    const button = this.getElement('button');
+    button.addEventListener('click', ()=>{
+      this.handller();
```
[the remark this answers is not in the document]
Comment (olha-khomiakova) [reply js/app/components/button/index.js:38]: Made a check.

File: js/app/components/component/index.js [hunk 7]
```diff
@@ -54,8 +54,14 @@ export class Component {
    * that was passed in the constructor when the component was created.
    */
   render() {
-    const tempElement = document.createElement('div');
-    tempElement.innerHTML = this.markup();
+    const markup = this.markup();
+    let tempElement;
+    if (markup.substring(0, markup.indexOf('>')+1).trim() === '<tr>') {
```
[the remark this answers is not in the document]
Comment (olha-khomiakova) [reply js/app/components/component/index.js:59]: Done.

File: js/app/file-list-page/file-list-item/index.js (hunk 1 above)
Comment (olha-khomiakova) [reply js/app/file-list-page/file-list-item/index.js:58]: Done.

File: js/app/file-list-page/file-list/index.js (hunk 2 above)
Comment (olha-khomiakova) [reply js/app/file-list-page/file-list/index.js:36]: Done.

File: js/app/file-list-page/link-with-action/index.js (hunk 3 above)
Comment (olha-khomiakova) [reply js/app/file-list-page/link-with-action/index.js:12]: Done.

File: js/app/file-list-page/link-with-action/index.js (hunk 4 above)
Comment (olha-khomiakova) [reply js/app/file-list-page/link-with-action/index.js]: Done.

**Review by dkalinin — APPROVED** [review]
image
Let's be consistent with the upper case:
"Create new directory"
"Upload file"
Please merge after fix.

**Pull request merged**